src/ceph-daemon

```diff
@@ -0,0 +1,1290 @@
+    subprocess.check_output(['rm', '-rf', data_dir])
+
+##################################
+
+def command_rm_cluster():
+    if not args.force:
+        raise RuntimeError('must pass --force to proceed: '
+                           'this command may destroy precious data!')
+
+    unit_name = 'ceph-%s.target' % args.fsid
+    try:
+        subprocess.check_output(['systemctl', 'stop', unit_name])
+        subprocess.check_output(['systemctl', 'disable', unit_name])
+    except subprocess.CalledProcessError:
+        pass
+
+    slice_name = 'system-%s.slice' % (
+        ('ceph-%s' % args.fsid).replace('-', '\\x2d'))
+    try:
+        subprocess.check_output(['systemctl', 'stop', slice_name])
+    except subprocess.CalledProcessError:
+        pass
+
+    # FIXME: stop + disable individual daemon units, too?
+
+    # rm units
+    subprocess.check_output(['rm', '-f', args.unit_dir +
+                             '/ceph-%s@.service' % args.fsid])
+    subprocess.check_output(['rm', '-f', args.unit_dir +
+                             '/ceph-%s.target' % args.fsid])
+    subprocess.check_output(['rm', '-rf',
+                  args.unit_dir + '/ceph-%s.target.wants' % args.fsid])
+    # rm data
+    subprocess.check_output(['rm', '-rf', args.data_dir + '/' + args.fsid])
+    # rm logs
+    subprocess.check_output(['rm', '-rf', args.log_dir + '/' + args.fsid])
+    subprocess.check_output(['rm', '-rf', args.log_dir +
+                             '/*.wants/ceph-%s@*' % args.fsid])
+
+
+##################################
+
+parser = argparse.ArgumentParser(
+    description='Bootstrap Ceph daemons with systemd and containers.',
+    formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+parser.add_argument(
+    '--image',
+    default=DEFAULT_IMAGE,
+    help='container image')
+parser.add_argument(
+    '--docker',
+    action='store_true',
+    help='use docker instead of podman')
+parser.add_argument(
+    '--data-dir',
+    default=DATA_DIR,
+    help='base directory for daemon data')
+parser.add_argument(
+    '--log-dir',
+    default=LOG_DIR,
+    help='base directory for daemon logs')
+parser.add_argument(
+    '--unit-dir',
+    default=UNIT_DIR,
+    help='base directory for systemd units')
+subparsers = parser.add_subparsers(help='sub-command')
+
+parser_version = subparsers.add_parser(
+    'version', help='get ceph version from container')
+parser_version.set_defaults(func=command_version)
+
+parser_ls = subparsers.add_parser(
+    'ls', help='list daemon instances on this host')
+parser_ls.set_defaults(func=command_ls)
+
+parser_adopt = subparsers.add_parser(
+    'adopt', help='adopt daemon deployed with a different tool')
+parser_adopt.set_defaults(func=command_adopt)
+parser_adopt.add_argument(
+    '--name', '-n',
+    required=True,
+    help='daemon name (type.id)')
+parser_adopt.add_argument(
+    '--style',
+    required=True,
+    help='deployment style (legacy, ...)')
```

Comment:
are other types expected in the future?

Reply:
yes: rook:v1 at least.  maybe the ceph-ansible container mode needs its own type, maybe not... hopefully i can make the legacy mode understand both normal and containerized systemd unit names in a graceful way.